File: csharp/Builder/NewCustomerProductBuilder.cs
namespace Builder;

public class NewCustomerProductBuilder : ProductBuilder
{
    ProductModel model = new ProductModel();

    public override void ApplyDiscount()
    {
        model.DiscountedPrice = model.UnitPrice * (decimal)0.90;
        model.DiscountApplied = true;
    }

    public override void GetProductData()
    {
        model.Id = 1;
        model.CategoryName = "Teknoloji";
        model.Name = "bilgisayar";
        model.UnitPrice = 30000;
    }

    public override ProductModel GetProductModel()
    {
        return model;
    }
}

File: java/builder/src/main/NewCustomerProductBuilder.java
package main;

import java.math.BigDecimal;

public class NewCustomerProductBuilder extends ProductBuilder{

    Product product = new Product();

    @Override
    public void getProductData() {
        product.id = 1;
        product.categoryName = "teknoloji";
        product.name = "bilgsayar";
        product.unitPrice = BigDecimal.valueOf(30000);
    }

    @Override
    public void applyDiscount() {
        product.discountedPrice = BigDecimal.valueOf(product.unitPrice.doubleValue()).multiply(BigDecimal.valueOf(0.90));
        product.discountApplied = true;
    }

    @Override
    public Product getProduct() {
        return product;
    }
}
